tools/src/file_format/config.rs

```diff
@@ -164,6 +164,7 @@ pub struct Config {
     pub trees: BTreeMap<String, TreeConfig>,
     pub mozsearch_path: String,
     pub config_repo_path: String,
+    pub url_map_path: Option<String>,
```

Comment:
Structurally this is a bit weird since the url_map_path is conceptually tree-specific ([URL_MAP_PATH is defined using INDEX_ROOT](https://searchfox.org/mozsearch/rev/51ae44c9d9a80a6a53628248f5e5184766c83f71/scripts/mkindex.sh#69) which [is extracted from the tree config](https://searchfox.org/mozsearch/rev/51ae44c9d9a80a6a53628248f5e5184766c83f71/scripts/load-vars.sh#15)), so I think it would make more sense for this to be attached to the tree config.

That said, I think it does make sense to land this as-is (but maybe with a TODO or XXX) to quiet the warnings since in practice it doesn't matter because output-file is only ever used at the granularity of a single tree at a time[1].  And then we can address this in some future enhancement as part of a more comprehensive cleanup where it might matter.

1: It's really only the web-server.rs and pipeline-server.rs that ever want the fully populated config structure.